inference-engine/ie_bridges/python/tests/test_IENetwork.py

```diff
@@ -158,6 +158,51 @@ def test_reshape():
     ie = IECore()
     net = ie.read_network(model=test_net_xml, weights=test_net_bin)
     net.reshape({"data": (2, 3, 32, 32)})
+    assert net.input_info["data"].input_data.shape == [2, 3, 32, 32]
+
+
+@pytest.mark.ngraph_dependent_test
+@pytest.mark.parametrize("shape, p_shape", [
+    ([1, 3, 22, 22], [1, 3, -1, 25]),
+    ([1, 3, 22, 22], [-1, -1, -1, -1]),
+    ([1, 3, -1, 25], [1, 3, 22, -1])
+])
+def test_reshape_with_partial_shape(device, shape, p_shape):
+    ie = IECore()
+    if device == "CPU":
+        if ie.get_metric(device, "FULL_DEVICE_NAME") == "arm_compute::NEON":
+            pytest.skip("Can't run on ARM plugin due-to ngraph")
+    from conftest import create_ngraph_function
+    import ngraph as ng
+    function = create_ngraph_function(shape)
+    net = ng.function_to_cnn(function)
+    net.reshape({"data": p_shape})
+    changedFunction = ng.function_from_cnn(net)
+    p_shape = ng.impl.PartialShape(p_shape)
+    assert changedFunction.get_parameters()[0].get_partial_shape().is_dynamic
+    assert changedFunction.get_results()[0].get_output_partial_shape(0).is_dynamic
+    assert function.get_parameters()[0].get_partial_shape().is_dynamic
+    assert function.get_results()[0].get_output_partial_shape(0).is_dynamic
+    assert changedFunction.get_parameters()[0].get_partial_shape() == p_shape
+    assert changedFunction.get_results()[0].get_output_partial_shape(0) == p_shape
+    assert function.get_parameters()[0].get_partial_shape() == p_shape
+    assert function.get_results()[0].get_output_partial_shape(0) == p_shape
+
+
+@pytest.mark.ngraph_dependent_test
+def test_incorrect_reshape(device):
+    ie = IECore()
+    if device == "CPU":
+        if ie.get_metric(device, "FULL_DEVICE_NAME") == "arm_compute::NEON":
+            pytest.skip("Can't run on ARM plugin due-to ngraph")
```

Comment:
I think better to use pytests.skipif instead skipping logic inside test
https://docs.pytest.org/en/6.2.x/reference.html#pytest-mark-skipif

Reply:
the problem is there that ARM is also marked as CPU and we need to distinguish it.
BTW @AlexeyLebedev1 prepared another PR #7392 for skipping several tests on ARM. As soon as it is merged, @AlexeyLebedev1 will update this tests too.